test/integration/provider/azure.go

```diff
@@ -34,60 +34,74 @@ func additionalResourcesCheck(
 	resourceGroup,
 	tagName,
 	tagValue string,
-) {
+) error {
 	ctx := context.TODO()
 
 	networkInterfaces, err := clients.GetNic().List(ctx, resourceGroup)
 	if err != nil {
 		fmt.Printf("List operation failed on NIC from resource group %s with error %s,", resourceGroup, err.Error())
 	}
+
 	for _, networkInterface := range networkInterfaces.Values() {
 		if value, ok := networkInterface.Tags[tagName]; ok && tagValue == *value {
 			fmt.Println(networkInterface.Name)
 			nicDeleteFuture, err := clients.GetNic().Delete(ctx, resourceGroup, *networkInterface.Name)
 			if err != nil {
 				fmt.Printf("Delete operation failed on NIC %s with error %s,", *networkInterface.Name, err.Error())
+				return err
 			}
 			if err = nicDeleteFuture.WaitForCompletionRef(ctx, clients.GetClient()); err != nil {
 				fmt.Printf("Delete operation failed on NIC %s with error %s,", *networkInterface.Name, err.Error())
+				return err
 			}
+			fmt.Printf("Deleted an orphan NIC %s,", *networkInterface.Name)
 		}
 	}
+	return nil
 }
 
 // deleteVolume deletes the specified volume on Azure
 func deleteVolume(
 	clients spi.AzureDriverClientsInterface,
 	resourceGroup,
 	VolumeID string,
-) {
+) error {
 	ctx := context.TODO()
 	diskDeleteFuture, err := clients.GetDisk().Delete(ctx, resourceGroup, VolumeID)
 	if err != nil {
 		fmt.Printf("Delete operation failed on Volume %s with error %s,", VolumeID, err.Error())
+		return err
 	}
 
 	if err = diskDeleteFuture.WaitForCompletionRef(ctx, clients.GetClient()); err != nil {
 		fmt.Printf("Delete operation failed on Volume %s with error %s,", VolumeID, err.Error())
+		return err
 	}
+	fmt.Printf("Deleted an orphan Disk %s,", VolumeID)
+	return nil
 
 }
 
 // deleteVM deletes the specified Virtual Machine on Azure
 func deleteVM(
 	clients spi.AzureDriverClientsInterface,
 	resourceGroup,
-	VMName string) {
+	VMName string) error {
 
 	ctx := context.TODO()
 	virtualMachineFuture, err := clients.GetVM().Delete(ctx, resourceGroup, VMName, pointer.BoolPtr(false))
 	if err != nil {
 		fmt.Printf("Delete operation failed on VM %s with error %s,", VMName, err.Error())
+		return err
 	}
 
 	if err = virtualMachineFuture.WaitForCompletionRef(ctx, clients.GetClient()); err != nil {
 		fmt.Printf("Delete operation failed on VM %s with error %s,", VMName, err.Error())
+		return err
 	}
+
+	fmt.Printf("Deleted an orphan VM %s,", VMName)
+	return nil
 }
 
 // getAvailableDisks fetches all the disks associated with the VMs of the target cluster
```

Comment:
I think we should change these docstrings of functions as well like for this it should be `// getAvailableDisks check for orphan disks associated with target cluster and triggers their deletion, returning the disks which couldn't be deleted`

Reply:
Of course! I think even the function names can be rather `getOrphaned<Resource>`


`// getAvailableDisks returns the list of orphaned disks which couldn't be deleted.`

Reply:
except the `getOrphanedMachines`, because there we don't try to delete the orphaned machines , they will be cleaned up in the AfterSuite.

Reply:
Indeed.